server/src/test/java/org/apache/druid/server/http/DataSourcesResourceTest.java

```diff
@@ -817,23 +805,6 @@ public void testMarkAsUsedNonOvershadowedSegmentsIntervalException()
     EasyMock.verify(segmentsMetadataManager, inventoryView, server);
   }
 
-  @Test
-  public void testMarkAsUsedNonOvershadowedSegmentsNoDataSource()
```

Comment:
Can we still keep this test and change the verification instead? i.e., instead of a 204 response code, I think the change should return a 200 with `"numChangedSegments: 0` in the response.

Reply:
kept test and updated.